app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt

```diff
@@ -116,49 +156,776 @@ class AdministratorControlsFragmentTest {
+          position = 2,
+          targetViewId = R.id.topic_update_on_wifi_switch
+        )
+      ).check(matches(isChecked()))
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.auto_update_topic_switch
+        )
+      ).check(matches(not(isChecked())))
+      onView(isRoot()).perform(orientationPortrait())
+      scrollToPosition(position = 2)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.topic_update_on_wifi_switch
+        )
+      ).check(matches(isChecked()))
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.auto_update_topic_switch
+        )
+      ).check(matches(not(isChecked())))
+    }
+  }
+
+  @Test
+  fun testAdministratorControlsFragment_clickWifiContainer_wifiSwitchIsChecked() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        profileId = internalProfileId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 2)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.topic_update_on_wifi_constraint_layout
+        )
+      ).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.topic_update_on_wifi_switch
+        )
+      ).check(matches(isChecked()))
+    }
+  }
+
+  @Test
+  fun testAdministratorControlsFragment_clickAutoUpdateContainer_autoUpdateSwitchIsChecked() {
     launch<AdministratorControlsActivity>(
       createAdministratorControlsActivityIntent(
-        0
+        profileId = internalProfileId
       )
     ).use {
       testCoroutineDispatchers.runCurrent()
-      onView(withId(R.id.edit_profiles_text_view)).perform(click())
-      intended(hasComponent(ProfileListActivity::class.java.name))
+      scrollToPosition(position = 2)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.auto_update_topic_constraint_layout
+        )
+      ).perform(click())
+      testCoroutineDispatchers.runCurrent()
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.auto_update_topic_switch
+        )
+      ).check(matches(isChecked()))
     }
   }
 
   @Test
-  fun testAdministratorControlsFragment_clickAppVersion_checkSendingTheCorrectIntent() {
+  fun testAdministratorControlsFragment_wifiSwitchIsNonClickable() {
```

Comment:
In that case, the condition is 'initialState' (and it's worth clarifying so that others don't make any assumptions about what state is necessary to produce the verified outcome).